src/main/java/com/rackspace/jenkins_nodepool/NodePoolState.java

```
@@ -0,0 +1,135 @@
+     */
+    private static final Logger LOG = java.util.logging.Logger.getLogger(NodePoolState.class.getName());
+
+    private final String stateString;
+
+    /**
+     * Creates a new NodePool state enum entity.
+     *
+     * @param stateString the NodePool state string value
+     */
+    NodePoolState(final String stateString) {
+        this.stateString = stateString;
+    }
+
+    /**
+     * Converts a NodePool state string into a NodePoolState enum object.
+     *
+     * @param stateString the string representation of the NodePool state
+     * @return a NodePoolState object based on the string value, otherwise returns null if there is not match of the
+     * specified string to a NodePoolState.
+     */
+    public static NodePoolState fromString(final String stateString) {
+        LOG.log(INFO, String.format("Processing NodePoolState.fromString(%s)", stateString));
+        // The 'in-use' state is a bit tricky in Java as we can't use a hyphen in the symbol name. This is how the
+        // nodepool authors defined it so we must accept it. All the other symbols are fine and we can use the default
+        // valueOf() routine.
+        // So, we use this routine to properly translate it from a string to the corresponding symbol.  We are generous
+        // and accept either all one word, with a hyphen or with an underscore for the in-use string - in all cases it
+        // maps to the proper IN_USE enum symbol.
+        if (stateString.equalsIgnoreCase("inuse") ||
+                stateString.equalsIgnoreCase("in-use") ||
+                stateString.equalsIgnoreCase("in_use")) {
+            return NodePoolState.IN_USE;
+        } else {
+            for (NodePoolState s : NodePoolState.values()) {
+                if (s.stateString.equalsIgnoreCase(stateString)) {
+                    return s;
+                }
+            }
+            LOG.log(WARNING, String.format("No NodePoolState match for string: %s", stateString));
```

This looks like it should throw, are you returning null instead so that all the calls don't need to be wrapped in try/catch?

---

@hughsaunders Been debating this one. 3-4 options on error handling:

1) return null - sometimes common, but crude and relentless
2) return some default or unknown state - not great as it hides the error
3) throw an unchecked exception (e.g. IllegalArgumentException - think this is the default behavior for valueOf() method which is built-in to enums)
4) throw a checked exception (annoying for calling code)

I'll update to throw IAE unless you object.

---

Resolved, added additional unit tests.